### PR TITLE
CLOUDP-91945: Add support for endpoints v1.5

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/mongodbatlas.go [hunk 1]
```diff
@@ -34,13 +34,11 @@ import (
...
-	APIPublicV1Path = "api/atlas/v1.0/" // APIPublicV1Path specifies the v1 api path
-	jsonMediaType   = "application/json"
-	plainMediaType  = "text/plain"
-	gzipMediaType   = "application/gzip"
-	libraryName     = "go-mongodbatlas"
+	defaultBaseURL = "https://cloud.mongodb.com/"
+	jsonMediaType  = "application/json"
+	plainMediaType = "text/plain"
+	gzipMediaType  = "application/gzip"
+	libraryName    = "go-mongodbatlas"
```
Comment (andreaangiolillo) [inline mongodbatlas/mongodbatlas.go:41]: I have removed `APIPublicV1Path` because we will use https://cloud.mongodb.com/ from now on

**Comment by gssbzn** [comment]
you can delete this https://github.com/mongodb/go-client-mongodb-atlas/blob/3f489d7a4eedf9521b32799a134c102d2fec01cc/mongodbatlas/mongodbatlas.go#L39

**Comment by andreaangiolillo** [comment]
@gssbzn I have deleted `APIPublicV1Path`

File: mongodbatlas/mongodbatlas.go (hunk 1 above)
Comment (gssbzn) [reply mongodbatlas/mongodbatlas.go:41]: you could've kept it and concatenate

**Review by gssbzn — APPROVED** [review]
LGTM this is going to need the same change on the ops manager client cc @themantissa

**Pull request merged**